.github/workflows/update-oam-defs.yml

```diff
@@ -0,0 +1,94 @@
+name: Meshery Kuma Pattern Components Generator
+on:
+  push:
+    branches:
+      - "master"
+jobs:
+  check_kuma_version:
+    name: Check Kuma Version
+    runs-on: ubuntu-latest
+    outputs:
+      NEW_KUMA_VERSION: ${{ steps.glrt.outputs.release }}
+      CURRENT_KUMA_VERSION: ${{ env.CURRENT_KUMA_VERSION }}
+    steps:
+      - name: Get latest release tag
+        id: glrt
+        uses: pozetroninc/github-action-get-latest-release@master
+        with:
+          repository: kumahq/kuma 
+          excludes: prerelease, draft
+      - name: Check out code
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - name: Get CURRENT_KUMA_VERSION from .env
+        run: |
+          cat .github/.env > $GITHUB_ENV
+  generator:
+    name: Generate and push OAM definitions
+    needs: [check_kuma_version]
+    if: ${{ needs.check_kuma_version.outputs.NEW_KUMA_VERSION  >  needs.check_kuma_version.outputs.CURRENT_KUMA_VERSION }}
+    env:
+      NEW_KUMA_VERSION: ${{ needs.check_kuma_version.outputs.NEW_KUMA_VERSION }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          ref: "master"
+      - name: Get KUMA manifests
+        run: |
+          mkdir ./kuma-crds
+          cd ./kuma-crds
+          curl --remote-name-all `curl -H "Accept: application/vnd.github.VERSION.raw" https://api.github.com/repositories/175425051/contents/deployments/charts/kuma/crds\?ref\=$NEW_KUMA_VERSION| jq 'map(.download_url) | .[]' | tr -d '"'`
+          cd ..
+          awk 'FNR==1 && NR>1 { printf("\n%s\n\n","---") } 1' kuma-crds/*.yaml > ~/kuma.yaml
+      - name: Build jsonschema util
+        run: |
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
+          export NVM_DIR="$HOME/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+          nvm install v14.15.3
+          nvm install-latest-npm
+          git clone https://github.com/layer5io/kubeopenapi-jsonschema util
+          cd util
+          npm i; npm i nexe -g
+          make linux
+      - name: Generate Workload definitions
+        run: |
+          mkdir -p ./templates/oam/workloads-$NEW_KUMA_VERSION
+          cd util
```

Comment:
Consider changing this in this way https://github.com/meshery/meshery-osm/blob/master/.github/workflows/update-oam-defs.yml#L66-L70